target/linux/ath79/nand/base-files/lib/upgrade/platform.sh

```diff
@@ -12,20 +12,26 @@ RAMFS_COPY_DATA='/etc/fw_env.config /var/lock/fw_printenv.lock'
 
 platform_do_upgrade_mikrotik_nand() {
 	CI_KERNPART=none
+
 	local fw_mtd=$(find_mtd_part kernel)
 	fw_mtd="${fw_mtd/block/}"
 	[ -n "$fw_mtd" ] || return
-	mtd erase kernel
-	tar xf "$1" sysupgrade-routerboard/kernel -O | nandwrite -o "$fw_mtd" -
 
+	local board_dir=$(tar tf "$1" | grep -m 1 '^sysupgrade-.*/$')
```

Comment:
These changes I took from https://github.com/openwrt/openwrt/pull/2790/commits/63eb68b8d79298386d60ff9f411bd1c9754b0dfa

Reply:
Of course, I'm aware that you just included my patch here without any reference.
The proper solution would be to cherry-pick my entire patch into your tree (separately), instead of including it here stripping off any explanation and authorship.

Reply:
I see "cherry-pick" is what I wanted, that is much better. I've pulled your commit in and rebased it prior to this one. Please let me know if that is now looking correct.

N.B. I have tested your change and it works on these boards.

Reply:
Thanks, that's how you do it.